x/gov/keeper/invariants.go

```diff
@@ -25,19 +25,19 @@ func AllInvariants(keeper Keeper, bk types.BankKeeper) sdk.Invariant {
 // deposit amounts held on store
 func ModuleAccountInvariant(keeper Keeper, bk types.BankKeeper) sdk.Invariant {
 	return func(ctx sdk.Context) (string, bool) {
-		var expectedDeposits sdk.Coins
+		var deposits sdk.Coins
 
 		keeper.IterateAllDeposits(ctx, func(deposit types.Deposit) bool {
-			expectedDeposits = expectedDeposits.Add(deposit.Amount...)
+			deposits = deposits.Add(deposit.Amount...)
 			return false
 		})
 
 		macc := keeper.GetGovernanceAccount(ctx)
 		balances := bk.GetAllBalances(ctx, macc.GetAddress())
-		broken := !balances.IsEqual(expectedDeposits)
+		broken := deposits.IsAllGT(balances)
```

Comment:
This smells bad. `deposits >= balances` because on a successful proposal
* deposit record is not deleted from the state
* but the deposit is refunded.

Hmm, can anyone remind me why we should return an old deposit if it was refunded?

Reply:
Earlier we are removing all the deposits except the active proposals' deposits from state and using the events to query when it is required. But now we are removing query by events, if we have to query them we have to depend on state only.

Reply:
we can update the way we compute `deposits` in invariants - we to not count deposits from passed proposals. 

But my question is still valid - do we need to query deposits which have been refunded?

Reply:
> do we need to query deposits which have been refunded?

@alexanderbez

Reply:
> do we need to query deposits which have been refunded?

if it is not required to query the refunded deposits, we no need to keep them in the state as well. We can simply delete them when a proposal passed or rejected

Note: we are refunding for rejected proposals as well
ref: https://github.com/cosmos/cosmos-sdk/blob/atheesh/remove-query-events/x/gov/abci.go#L50-L54
https://github.com/cosmos/cosmos-sdk/blob/atheesh/remove-query-events/x/gov/keeper/tally.go#L97-L124